src/parenttext/canto.py

```diff
@@ -152,12 +154,15 @@ def download(client, path_template, asset: MediaAsset, destination):
     print(f"Download completed, path={dst}")
 
 
-if __name__ == "__main__":
-    with open("config.json", "r") as fh:
+def main(destination: str, config_file: str | None = None):
+    with open(config_file or "config.json", "r") as fh:
         config = json.load(fh)["sources"]["media_assets"]
 
     _env = Environment(undefined=ChainableUndefined)
-    load_dotenv(".env")
+    if not load_dotenv(".env"):
+        raise FileNotFoundError(
+            "Must have a .env file in current working directory or parent directories"
```

Comment:
Applies to current working directory only because specifying a location switches off the behaviour that attempts to find '.env' in parent directories.